BootloaderCommonPkg/Include/Guid/PayloadKeyHashGuid.h

```diff
@@ -14,21 +14,23 @@
 ///
 extern EFI_GUID gPayloadKeyHashGuid;
 
-#ifndef    SHA256_DIGEST_SIZE
-#define    SHA256_DIGEST_SIZE   32
-#endif
-#define    MAX_KEY_DIGEST_COUNT 3
 
+#pragma pack(1)
 typedef struct {
-  UINT32               ComponentType;
-  UINT8                Digest[SHA256_DIGEST_SIZE];
-} KEY_HASH_ITEM;
+  UINT32           Usage;

```

Comment:
What are the valid values for Usage?
Let's add a note in the code so that person reading the code can easily understand it.

Reply:
Currently **Usage** bits corresponds to below components. There would be more than one hash for a component as Public Keys.

#define  COMP_TYPE_STAGE_1B            0
#define  COMP_TYPE_STAGE_2             1
#define  COMP_TYPE_PAYLOAD             2
#define  COMP_TYPE_FIRMWARE_UPDATE     3
#define  COMP_TYPE_PUBKEY_CFG_DATA     4
#define  COMP_TYPE_PUBKEY_FWU          5
#define  COMP_TYPE_PUBKEY_OS           6

Reply:
A more clear method is to use typedef enum {}  for such kind of list. But the enum type size depends on compiler although there are some options to make size more specific (e.g. add a big unused number when define enum to use UINT32, or use -fshort-enums for gcc to be one byte if possible) . Later we could explore it further.
For this patch,  how about we rename it clearly as CompType and use UINT8 instead of UINT32 to make it more clear.
and if possible, move these COMP TYPE definition to this file and let SecureBootLib.h include this file, make sense?

Reply:
Maybe we could use following definition. It uses UINT8 and leverage compiler to have a data type check.
//
// Hash Table Definition for Component and Publick Key 
//
typedef UINT8   HASH_TABLE_COMP_TYPE;
#define COMP_TYPE_STAGE_1B 0
#define COMP_TYPE_STAGE_2   1
... ...
typedef struct {
  HASH_TABLE_COMP_TYPE  HashCompType;
  UINT8                                 HashAlg;
  UINT16                               DigestLen;
  UINT8                                 Digest[0];
} HASH_STORE_DATA;

Reply:
After sync with Maurice and Subash, got to know the usage is the bitmask usage for hashes. So the final change would be similar as below:
//
// Hash Table Definition for Component and Publick Key Usage
//
typedef UINT32  HASH_COMP_USAGE;
#define HASH_USAGE_STAGE_1B  BIT0
#define HASH_USAGE_STAGE_2   BIT1
#define HASH_USAGE_PAYLOAD  BIT2
#define HASH_USAGE_FIRMWARE_UPDATE  BIT3
#define HASH_USAGE_ROOT_KEY  BIT8
#define HASH_USAGE_PUBKEY_CFG_DATA   BIT9
#define HASH_USAGE_PUBKEY_FWU      BIT10
#define HASH_USAGE_PUBKEY_OS        BIT11
typedef struct {
HASH_COMP_USAGE  HashCompUsage;
UINT8 HashAlg;
UINT8    Reserved;
UINT16 DigestLen;
UINT8 Digest[0];
} HASH_STORE_DATA;

Reply:
Thanks Guo. I have moved hash component definitions to keyHashGuid.h.
Regarding bitwise, added reference definitions for usage. These definitions wouldn't consumed now as we need to change many API definitions.